coops-ui/src/components/Common/OfficeAddresses.vue

```diff
@@ -0,0 +1,813 @@
+          </li>
+        </div>
+        <div v-else>
+          <li class="container" v-if="!showAddressForm">
+            <div class="meta-container">
+              <label>Record Office</label>
+              <div class="meta-container__inner">
+                <span>
+                  Same as Registered Office
+                </span>
+              </div>
+            </div>
+          </li>
+        </div>
+      </div>
+      <!-- Form Btn Section -->
+      <li>
+        <div
+          class="form__row form__btns"
+          v-show="showAddressForm"
+        >
+          <v-btn
+            class="update-btn"
+            color="primary"
+            id="reg-off-update-addr-btn"
+            :disabled="!formValid"
+            @click="updateAddress"
+          >
+            Update Addresses
+          </v-btn>
+          <v-btn id="reg-off-cancel-addr-btn" @click="cancelEditAddress">Cancel</v-btn>
+        </div>
+      </li>
+    </ul>
+  </v-card>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Emit, Prop, Watch, Mixins } from 'vue-property-decorator'
+import isEmpty from 'lodash.isempty'
+
+// Schemas
+import { addressSchema } from '@/schemas'
+
+// Components
+import BaseAddress from 'sbc-common-components/src/components/BaseAddress.vue'
+
+// Mixins
+import { AddressMixin, CommonMixin, EntityFilterMixin } from '@/mixins'
+
+// Interfaces
+import { BaseAddressObjIF, BcorpAddressIf, AddressIF } from '@/interfaces/address-interfaces'
+
+// Constants
+import { ADDRESSCHANGED } from '@/constants'
+
+// Enums
+import { EntityTypes } from '@/enums'
+
+@Component({
+  components: {
+    'delivery-address': BaseAddress,
+    'mailing-address': BaseAddress
+  }
+})
+export default class OfficeAddresses extends Mixins(AddressMixin, CommonMixin, EntityFilterMixin) {
+    /**
+     * Indicates whether the change button should be disabled or not
+     */
+    @Prop({ default: false })
+    readonly changeButtonDisabled: boolean
+
+    /**
+     * Addresses object from the parent page.
+     * If this is null then this is a new filing; otherwise these are the addresses from a draft filing.
+     * This will be emitted back to the parent page when the addresses are updated.
+     */
+    @Prop({ default: null })
+    readonly addresses: BcorpAddressIf | null
+
+    /**
+     * Registered Office address object passed in from the parent
+     */
+    @Prop({ default: null })
+    private registeredAddress: BaseAddressObjIF
+
+    /**
+     * Records Office address object passed in from the parent
+     */
+    @Prop({ default: null })
+    private recordsAddress: BaseAddressObjIF
+
+    // The two addresses that come from the store. These are used to reset the address.
+    private deliveryAddressOriginal = {} as AddressIF
+    private mailingAddressOriginal = {} as AddressIF
+
+    // The two addresses that come from the store. These are used to reset the address.
+    private recDeliveryAddressOriginal = {} as AddressIF
```

Comment:
Any idea why we need two separate address properties to reset the address? If there's a reason then please add comments here to discern these. Otherwise I'm confused. If there's no reason then please, please, please simplify this code. Thanks either way!

Reply:
Out of scope.